### PR TITLE
Modulized UI Components in InputView

### PR DESCRIPTION
**Pull request by WhiredPlanck**
#### Issue tracker
Fixes will automatically close the related issues

Fixes # N/A

#### Feature
Describe features of pull request

This PR is going to improve the management of the UI components showed in InputView, following:

1. Split out the bar UIs into QuickBar component and split out the keyboard UIs into KeyboardWindow component.
2. Use ViewAnimator to make life easier to switch UI state, free people from setting views' visibility as GONE/INVISIBLE here or as VISIBLE there.
3. Introduce [Koin](https://insert-koin.io), a lightweight Kotlin dependency injection framework, to automatically manage the dependencies between these UI components, avoiding complex template codes and long construct arguments.

These are just the beginning, and there will be many more in the future.

#### Code of conduct
- [x] [CONTRIBUTING](CONTRIBUTING.md)

#### Style lint
- [x] `make sytle-lint`

#### Build pass
- [x] `make debug`

#### Manually test
- [x] Done

#### Code Review
1. No wildcards import
5. Manual build and test pass
6. GitHub action ci pass
7. At least one contributor reviews and votes
8. Can be merged clean without conflicts
9. PR will be merged by rebase upstream base

#### Daily build
Login and download artifact at https://github.com/osfans/trime/actions

### CODE REVIEW

File: app/src/main/java/com/osfans/trime/ime/core/Trime.java [hunk 1]
```diff
@@ -332,10 +328,9 @@ public void selectLiquidKeyboard(final int tabIndex) {
       symbolKeyboardType = SymbolKeyboardType.NO_KEY;
       // 设置液体键盘处于隐藏状态
       TabManager.get().setTabExited();
-      symbolInput.setVisibility(View.GONE);
+      inputView.switchUiByIndex(0);
```
Comment (MokOopsing) [inline app/src/main/java/com/osfans/trime/ime/core/Trime.java]: 建议用enum而不是魔鬼数字，不然现在理好了，后面的人改起来不小心把1写成0，也不好看出差别，容易引入bug

**Comment by goofyz** [comment]
I assume you will move `startKoin()` related code to `Application.onCreate()` and `module{ ... }` to a separate package (e.g. `di`) later so they will be better organized.

**Review by goofyz — APPROVED** [review]
And I think that using koin with annotation will be better as it will have compile time safety.

**Comment by WhiredPlanck** [comment]
> I assume you will move `startKoin()` related code to `Application.onCreate()` and `module{ ... }` to a separate package (e.g. `di`) later so they will be better organized.

That's hard. Because to ensure injecting the initialized instances (especially the context and service) to these components, I think I may only startKoin when init InputView.

> And I think that using koin with annotation will be better as it will have compile time safety.

Annotations need ksp to generate code, which is too much I think for current usage.

**Pull request merged**